src/System.Linq.Expressions/tests/BinaryOperators/Arithmetic/BinaryPowerTests.cs

```diff
@@ -0,0 +1,427 @@
+        {
+            Expression<Func<float>> e =
+                Expression.Lambda<Func<float>>(
+                    Expression.Power(
+                        Expression.Constant(a, typeof(float)),
+                        Expression.Constant(b, typeof(float)),
+                        typeof(BinaryPowerTests).GetTypeInfo().GetDeclaredMethod("PowerFloat")
+                    ));
+            Func<float> f = e.Compile(useInterpreter);
+
+            Assert.Equal(PowerFloat(a, b), f());
+        }
+
+        private static void VerifyDoublePower(double a, double b, bool useInterpreter)
+        {
+            Expression<Func<double>> e =
+                Expression.Lambda<Func<double>>(
+                    Expression.Power(
+                        Expression.Constant(a, typeof(double)),
+                        Expression.Constant(b, typeof(double)),
+                        typeof(BinaryPowerTests).GetTypeInfo().GetDeclaredMethod("PowerDouble")
+                    ));
+            Func<double> f = e.Compile(useInterpreter);
+
+            Assert.Equal(PowerDouble(a, b), f());
+        }
+
+        private static void VerifyDecimalPower(decimal a, decimal b, bool useInterpreter)
+        {
+            Expression<Func<decimal>> e =
+                Expression.Lambda<Func<decimal>>(
+                    Expression.Power(
+                        Expression.Constant(a, typeof(decimal)),
+                        Expression.Constant(b, typeof(decimal)),
+                        typeof(BinaryPowerTests).GetTypeInfo().GetDeclaredMethod("PowerDecimal")
+                    ));
+            Func<decimal> f = e.Compile(useInterpreter);
+
+            decimal expected = 0;
+            try
+            {
+                expected = PowerDecimal(a, b);
+            }
+            catch (OverflowException)
+            {
+                Assert.Throws<OverflowException>(() => f());
+                return;
+            }
+
+            Assert.Equal(expected, f());
+        }
+
+        private static void VerifyCharPower(char a, char b, bool useInterpreter)
+        {
+            Expression<Func<char>> e =
+                Expression.Lambda<Func<char>>(
+                    Expression.Power(
+                        Expression.Constant(a, typeof(char)),
+                        Expression.Constant(b, typeof(char)),
+                        typeof(BinaryPowerTests).GetTypeInfo().GetDeclaredMethod("PowerChar")
+                    ));
+            Func<char> f = e.Compile(useInterpreter);
+
+            Assert.Equal(PowerChar(a, b), f());
+        }
+
+        [Fact]
+        public static void ToStringTest()
+        {
+            var e = Expression.Power(Expression.Parameter(typeof(double), "a"), Expression.Parameter(typeof(double), "b"));
+            Assert.Equal("(a ^ b)", e.ToString()); // QUIRK: Same output as ExclusiveOr, see https://github.com/dotnet/corefx/issues/11415
```

Comment:
This is an interesting quirk. Note that `PowerAssign` uses `**=`, so it'd be logical for this to be `**`. Depending on the compatibility stance for a rarely used node type (though it is a 3.5 generation node; I assume VB emits it), we could decide to change it or not; see #11415.